evap/staff/views.py

```diff
@@ -1141,7 +1153,11 @@ def user_import(request):
 
 @staff_required
 def user_edit(request, user_id):
-    user = get_object_or_404(UserProfile, id=user_id)
+    try:
+        user = UserProfile.objects.get(id=user_id)
+    except UserProfile.DoesNotExist:
+        raise Http404('No UserProfile matches the given query.')
```

Comment:
have you checked whether get_object_or_404 can be used again since we're overriding the default manager?